### PR TITLE
Don't add replica labels to exemplar labels, only to its series labels

### LINKED ISSUE #4222: Replica labels shouldn't be added to exemplar labels
**Thanos, Prometheus and Golang version used**:

Latest

**What happened**:

While working on #4221, I found for exemplar APIs, replica labels are added to both series labels and exemplar labels.

For example, one example response is:

```
curl 'localhost:49463/api/v1/query_exemplars?end=1620780122.472&query=http_request_duration_seconds_bucket%7Bhandler%3D%22label_names%22%7D&start=1620772922.472' | jq .

{
  "status": "success",
  "data": [
    {
      "seriesLabels": {
        "__name__": "http_request_duration_seconds_bucket",
        "code": "200",
        "handler": "label_names",
        "instance": "e2e_test_exemplars_fanout-querier-query:8080",
        "job": "myself",
        "le": "0.01",
        "method": "get",
        "prometheus": "ha"
      },
      "exemplars": [
        {
          "labels": {
            "prometheus": "ha",
            "traceID": "69ab3ca4a0b7f639"
          },
          "timestamp": 1620776522.474,
          "value": "0.001569066"
        }
      ]
    }
  ]
}
```

Here `prometheus=ha` is the external label. It should be added to the series labels, but not to exemplar labels.

**What you expected to happen**:
`prometheus=ha` appears in series labels and doesn't appear in exemplar labels.

### PR DESCRIPTION
**Pull request by yeya24**
Signed-off-by: yeya24 <yb532204897@gmail.com>

* [x] I added CHANGELOG entry for this change.
* [ ] Change is not relevant to the end user.

## Changes

Fixes #4222 

This pr removes the step of adding replica labels to exemplars. Only adding replica labels to exemplars' series labels.

### CODE REVIEW

File: pkg/exemplars/exemplars.go [hunk 1]
```diff
@@ -79,15 +79,11 @@ func (rr *GRPCClient) Exemplars(ctx context.Context, req *exemplarspb.ExemplarsR
 		return nil, nil, errors.Wrap(err, "proxy Exemplars")
 	}
 
-	resp.data = dedupExemplarsData(resp.data, rr.replicaLabels)
-	for _, d := range resp.data {
-		d.Exemplars = dedupExemplars(d.Exemplars, rr.replicaLabels)
-	}
-
+	resp.data = dedupExemplarsSeriesLabels(resp.data, rr.replicaLabels)
```
Comment (GiedriusS) [inline pkg/exemplars/exemplars.go:82]: I'm a bit confused. You say:

> This pr removes the step of adding replica labels to exemplars. Only adding replica labels to exemplars' series labels.

`dedupExemplarsSeriesLabels` deduplicates the data using the replica labels and removes those labels. Here now you are now only doing it on the series so series will be without the replica labels, and exemplars - will be with the replica labels. Shouldn't it be the other way around i.e. the deduplication should only happen on the exemplars? :smile:

Comment (yeya24) [reply pkg/exemplars/exemplars.go:82]: Sorry, that was my big mistake. Thanks for catching it. I should modify the code from the Prometheus proxy side.

Comment (yeya24) [reply pkg/exemplars/exemplars.go]: Updated. PTAL

**Comment by yeya24** [comment]
Fixed #4234 in da51e79a47d34ad56aeaa949f0a1819958d7cdd5

**Review by onprem — APPROVED** [review]
LGTM, thanks!

**Review by GiedriusS — APPROVED** [review]
LGTM

**Review by vovanloc0798 — COMMENTED** [review]
https://coinsniper.net/coin/611

**Pull request merged**